conan/tools/gnu/pkgconfig.py

```diff
@@ -0,0 +1,93 @@
+import subprocess
+
+from conan.tools.env import Environment
+from conans.errors import ConanException
+from conans.util.runners import check_output_runner
+
+
+class PkgConfig:
+
+    def __init__(self, conanfile, package, pkg_config_path=None):
+        """
+        :param package: library (package) name, such as libastral
+        """
+        self._conanfile = conanfile
+        self._package = package
+        self._info = {}
+        self._pkg_config_path = pkg_config_path
+        self._variables = None
+
+    def _parse_output(self, option):
+        executable = self._conanfile.conf["tools.gnu:pkg_config"] or "pkg-config"
+        command = [executable, '--' + option, self._package, '--print-errors']
+        try:
+            env = Environment()
+            if self._pkg_config_path:
+                env.prepend_path("PKG_CONFIG_PATH", self._pkg_config_path)
+            with env.vars(self._conanfile).apply():
+                return check_output_runner(command).strip()
+        except subprocess.CalledProcessError as e:
+            raise ConanException('pkg-config command %s failed with error: %s' % (command, e))
+
+    def _get_option(self, option):
+        if option not in self._info:
+            self._info[option] = self._parse_output(option)
+        return self._info[option]
+
+    @property
+    def includedirs(self):
+        return [include[2:] for include in self._get_option('cflags-only-I').split()]
+
+    @property
+    def cflags(self):
+        return [flag for flag in self._get_option('cflags-only-other').split()
+                if not flag.startswith("-D")]
+
+    @property
+    def defines(self):
+        return [flag for flag in self._get_option('cflags-only-other').split()
+                if flag.startswith("-D")]
+
+    @property
+    def libdirs(self):
+        return [lib[2:] for lib in self._get_option('libs-only-L').split()]
+
+    @property
+    def libs(self):
+        return [lib[2:] for lib in self._get_option('libs-only-l').split()]
```

Comment:
I guess that this `.split()` will fail in the case of having paths with spaces. Could you add a test using that?

Reply:
This goes for all `.split()`'s in this class.
Perhaps `shlex.split` can come to the rescue?

Reply:
I think if we agree on the interfaces and general approach and behavior, we can move this PR, and I suggest @franramirez688 doing some extra tests with path with spaces and fix the implementation as necessary (this was apparently not covered by previous functionality either)

Reply:
Fine for me